ast/builtins.go

```diff
@@ -1317,6 +1318,22 @@ var StringReverse = &Builtin{
 	Categories: stringsCat,
 }
 
+var RenderTemplate = &Builtin{
+	Name: "strings.render_template",
+	Description: `Renders a templated string with given template variables injected. For a given template string and key/value mapping, values will be injected into the template where they are referenced by key. 
+	The templated string can contain simple syntax (such as "Hello {{.name}}") or complex syntax (such as "{{if .shouldsayhello}}Hello world!{{end}}").
+	For more examples, see https://pkg.go.dev/text/template 
+	`,
+	Decl: types.NewFunction(
+		types.Args(
+			types.Named("str", types.S).Description("a templated string"),
+			types.Named("vars", types.A).Description("a mapping of template variable keys to values"),
```

Comment:
Shouldn't this be an object (`types.NewObject`)?

Reply:
Yes, thanks for the suggestion